aptos-move/block-executor/src/executor.rs

```diff
@@ -492,25 +500,34 @@ where
         );
         let execution_result = executor.execute_transaction(&sync_view, txn, idx_to_execute);
 
-        let mut prev_modified_resource_keys = last_input_output
-            .modified_resource_keys(idx_to_execute)
-            .map_or_else(HashSet::new, |keys| keys.map(|(k, _)| k).collect());
-        let mut prev_modified_group_keys: HashMap<T::Key, HashSet<T::Tag>> = last_input_output
-            .modified_group_key_and_tags_cloned(idx_to_execute)
-            .into_iter()
-            .collect();
-
         let mut read_set = sync_view.take_parallel_reads();
         if read_set.is_incorrect_use() {
-            return Err(PanicOr::from(code_invariant_error(format!(
+            return Err(code_invariant_error(format!(
                 "Incorrect use detected in CapturedReads after executing txn = {} incarnation = {}",
                 idx_to_execute, incarnation
-            ))));
+            )));
         }
-
         let processed_output =
             Self::process_execution_result(&execution_result, &mut read_set, idx_to_execute)?;
 
+        // CAUTION: any updates applied to shared multi-versioned data structures must occur
+        // below, without any short circuits. To enforce the invariant that the output from a
```

Comment:
PanicError is fine since that falls back, this means normal execution flow short circuits, I probably should clarify.